src/kube.rs

```diff
@@ -0,0 +1,76 @@
+use kube::{
+    api::{Patch, PatchParams},
+    core::{DynamicObject, GroupVersionKind, TypeMeta},
+    discovery::Scope,
+    Api, Client, Discovery, ResourceExt,
+};
+use serde::Deserialize;
+use snafu::{ResultExt, Snafu};
+
+#[derive(Debug, Snafu)]
+pub enum KubeError {
+    #[snafu(display("kube error: {source}"))]
+    KubeError { source: kube::error::Error },
+
+    #[snafu(display("yaml error: {source}"))]
+    YamlError { source: serde_yaml::Error },
+
+    #[snafu(display("object type error: {msg}"))]
+    ObjectTypeError { msg: String },
+
+    #[snafu(display("discovery error: {msg}"))]
+    DiscoveryError { msg: String },
+}
+
+pub async fn deploy_manifests(manifests: &str, namespace: &str) -> Result<(), KubeError> {
+    let client = Client::try_default().await.context(KubeSnafu {})?;
+    let discovery = Discovery::new(client.clone())
+        .run()
+        .await
+        .context(KubeSnafu {})?;
+
+    for manifest in serde_yaml::Deserializer::from_str(manifests) {
+        let mut object = DynamicObject::deserialize(manifest).context(YamlSnafu {})?;
+        let object_type = object.types.as_ref().ok_or(
+            ObjectTypeSnafu {
+                msg: format!(
+                    "failed to deploy manifest because type of object {object:?} is not set"
+                ),
+            }
```

Comment:
This error formatting should be in the `#[snafu(display)]`

Reply:
Changed in df0076f.